### PR TITLE
Slayer

### PR DESCRIPTION
**Pull request by acbramley**
giphy

### CODE REVIEW

File: slayer/main.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	"k8s.io/client-go/rest"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"k8s.io/api/core/v1"
+)
+
+var (
+	cliGithubToken = kingpin.Flag("githubToken", "The Github access token.").Required().String()
```
Comment (nicksantamaria) [inline slayer/main.go]: Can we update this to `--token` and add environment variable default of `GITHUB_TOKEN`

File: slayer/main.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			panic(err.Error())
+		}
+	}
+}
+
+// getPodsToSlay gathers any pods with closed PRs.
+func getPodsToSlay(pods *v1.PodList, clientGithub *github.Client, ctx context.Context) ([]v1.Pod, error) {
+	var podsToSlay []v1.Pod
+	for _, pod := range pods.Items {
+		if _, ok := pod.Annotations[KeyCircleProjectRepoName]; !ok {
```
Comment (nicksantamaria) [inline slayer/main.go]: Replace `KeyCircleProjectRepoName` with `cmd.metadata.AnnotationCircleCIRepositoryName`

File: slayer/main.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			fmt.Printf("Pod %s missing the %s annotation, skipping\n", pod.ObjectMeta.Name, KeyCircleProjectRepoName)
```
Comment (nicksantamaria) [inline slayer/main.go]: Replace `KeyCircleProjectRepoName` with `cmd.metadata.AnnotationCircleCIRepositoryName`

File: slayer/main.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+		if _, ok := pod.Annotations[KeyCircleProjectUserName]; !ok {
+			fmt.Printf("Pod %s missing the %s annotation, skipping\n", pod.ObjectMeta.Name, KeyCircleProjectUserName)
+			continue
+		}
+		if _, ok := pod.Annotations[KeyCircleBranch]; !ok {
+			fmt.Printf("Pod %s missing the %s annotation, skipping\n", pod.ObjectMeta.Name, KeyCircleBranch)
+			continue
+		}
+
+		prs, err := getClosedPrs(clientGithub, ctx, pod.Annotations[KeyCircleProjectUserName], pod.Annotations[KeyCircleProjectRepoName])
```
Comment (nicksantamaria) [inline slayer/main.go]: Replace `KeyCircleProjectRepoName` with `cmd.metadata.AnnotationCircleCIRepositoryName`

File: slayer/main.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+// getPodsToSlay gathers any pods with closed PRs.
+func getPodsToSlay(pods *v1.PodList, clientGithub *github.Client, ctx context.Context) ([]v1.Pod, error) {
+	var podsToSlay []v1.Pod
+	for _, pod := range pods.Items {
+		if _, ok := pod.Annotations[KeyCircleProjectRepoName]; !ok {
+			fmt.Printf("Pod %s missing the %s annotation, skipping\n", pod.ObjectMeta.Name, KeyCircleProjectRepoName)
+			continue
+		}
+		if _, ok := pod.Annotations[KeyCircleProjectUserName]; !ok {
```
Comment (nicksantamaria) [inline slayer/main.go]: Replace `KeyCircleProjectUserName` with `cmd.metadata.AnnotationCircleCIRepositoryUsername`

File: slayer/main.go [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+	"gopkg.in/alecthomas/kingpin.v2"
+	"k8s.io/api/core/v1"
+)
+
+var (
+	cliGithubToken = kingpin.Flag("githubToken", "The Github access token.").Required().String()
+	prCache        map[string][]*github.PullRequest
+)
+
+const (
```
Comment (nicksantamaria) [inline slayer/main.go]: Remove these constants and replace them in code with the `cmd.metadata.Annotation...` constants.

File: slayer/main.go [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+	for _, pod := range pods.Items {
+		if _, ok := pod.Annotations[KeyCircleProjectRepoName]; !ok {
+			fmt.Printf("Pod %s missing the %s annotation, skipping\n", pod.ObjectMeta.Name, KeyCircleProjectRepoName)
+			continue
+		}
+		if _, ok := pod.Annotations[KeyCircleProjectUserName]; !ok {
+			fmt.Printf("Pod %s missing the %s annotation, skipping\n", pod.ObjectMeta.Name, KeyCircleProjectUserName)
+			continue
+		}
+		if _, ok := pod.Annotations[KeyCircleBranch]; !ok {
```
Comment (nicksantamaria) [inline slayer/main.go]: Replace `KeyCircleBranch` with `cmd.metadata.AnnotationCircleCIBranch`

**Comment by nickschuch** [comment]
TIME TO SLAY!

**Pull request merged**